torchtext/experimental/datasets/raw/translation.py

```diff
@@ -0,0 +1,391 @@
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/test_2016.4.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/test_2016.5.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/test_2016.5.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.1.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.1.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.2.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.2.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.3.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.3.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.4.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.4.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.5.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/train.5.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.1.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.1.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.2.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.2.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.3.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.3.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.4.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.4.en.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.5.de.gz",
+        "https://raw.githubusercontent.com/multi30k/dataset/master/data/task2/raw/val.5.en.gz"
+    ],
+    'WMT14':
+    'https://drive.google.com/uc?export=download&id=0B_bZck-ksdkpM25jRUN2X2UxMm8',
+    'IWSLT':
+    'https://wit3.fbk.eu/archive/2016-01//texts/{}/{}/{}.tgz'
+}
+
+
+def _read_text_iterator(path):
+    with io.open(path, encoding="utf8") as f:
+        reader = unicode_csv_reader(f)
+        for row in reader:
+            yield " ".join(row)
+
+
+def _clean_xml_file(f_xml):
+    f_txt = os.path.splitext(f_xml)[0]
+    with codecs.open(f_txt, mode='w', encoding='utf-8') as fd_txt:
+        root = ET.parse(f_xml).getroot()[0]
+        for doc in root.findall('doc'):
+            for e in doc.findall('seg'):
+                fd_txt.write(e.text.strip() + '\n')
+
+
+def _clean_tags_file(f_orig):
```

Comment:
what's this func for?

Reply:
There are files such as train.tags.en-de.en, train.tags.en-cs.en, etc which contains something similar to xml files. The contents look like the following

```
 <translator href="http://www.ted.com/profiles/383709">Abdulatif Mahgoub</translator>
 <reviewer href="http://www.ted.com/profiles/122540">Anwar Dafa-Alla</reviewer>
 <url>http://www.ted.com/talks/steve_keil_a_manifesto_for_play_for_bulgaria_and_beyond</url>
 <keywords>talks, Europe, TEDx, business, culture, economics, play, social change</keywords>
 <speaker>Steve Keil</speaker>
 <talkid>1170</talkid>
 <title>ستيف كايل: بيان رسمي للعب ، الى بلغاريا و ماوراءها</title>
 <description>TED Talk Subtitles and Transcript: في تيدأكس بيىجي في صوفيا، بلغاريا ، في صوفيا ، يحارب ستيف كايل "المعلومات الثقافية المتوارثة" والتي أصابت وطنه بلغاريا - ويدعو للعودة الى اللعب لتنشيط الاقتصاد، التعليم ، والمجتمع. حديث رائع ذو رسالة عالمية للأفراد في كل مكان ، الذين يعيدون ابتكار اماكن عملهم ، مدارسهم ، حياتهم.</description>
  إنني هنا اليوم  لأبدأ ثورة. 
  والآن وقبل أن تهبوا،  أو تبدأوا بالغناء،  أو اختيار لون مفضل،  أريد تحديد ما أعنيه بثورة. 
  بكلمة ثورة،  أعني تغييرا جذريا وبعيد المدى  في طريقة تفكيرنا وتصرفاتنا--  في طريقة تفكيرنا وطريقة تصرفاتنا. 
  والآن لماذا ، ستيف، لماذا نحتاج الى ثورة؟ 
  نحتاج إلى ثورة  لأن الأشياء لا تعمل؛ انها لا تعمل تماماً. 
  وهذا يجعلني حزينا جدا،  لأنني سئمت وتعبت من الأشياء التي لا تعمل. 
  أوتعرفون، لقد سئمت وتعبت منّا ونحن لا نحيا بامكانياتنا. 
  سئمت وتعبت منّا وقد أصبحنا متأخرين. 
```

Reply:
Sorry for the lack of context. Those tags files are parts of IWSLT dataset when you download them